### PR TITLE
Flak Cannon from Unreal Tournament [WIP]

### PR DESCRIPTION
**Pull request by ghost**
image
image

:cl: zamolxius
add: Added the flak cannon for traitor engineers, a kind of shotgun that fires ricocheting metal shards when fed with metal sheets
/:cl:

**work in progress**
I plan on it shooting a bunch of metal shards usually, and upon using it inhand it switches to a mode that shoots a slow-moving ball that explodes into a few more shards from where it lands, and does more damage on a direct hit.

https://unreal.fandom.com/wiki/Flak_Cannon

### CODE REVIEW

File: hippiestation/code/game/objects/items/weaponry.dm [hunk 1]
```diff
@@ -231,3 +231,111 @@
...
+
+
+/obj/item/gun/ballistic/flak_cannon/attackby(obj/item/A, mob/user, params)
+	if(istype(A,/obj/item/stack/sheet/metal))
+		if(A.use(10))
+			to_chat(user, "<span class='notice'>The [src] gobbles up the metal.</span>")
+		else
+			to_chat(user, "<span class='warning'>The intake will not accept any less than ten sheets !</span>")
+			return
+
```
Comment (ghost) [inline hippiestation/code/game/objects/items/weaponry.dm]: gotta find a way to load the thing up and switch between the flak shot and the flak ball magazines at will, while preserving the ammo count and (dividing ammo count by 3 when going from shot to ball, multiplying by 3 when going from ball to shot) I believe it's stored as objects though, not integers, in the magazine? 
and to create new ammo in the magazine when you click the gun with metal sheets, using up 10 sheets in flak shot mode to give 1 charge and 30 sheets in ball mode to give 1 charge

**Comment by yoyobatty** [comment]
How's the progress?

**Comment by ghost** [comment]
think i should just make it energy based and recharge cell when using metal on it,  not allow recharger charging, since energy weapons already have multiple firing modes so

**Comment by Manly-Dwarf** [comment]
could be a very cool and promising improvised weapon

**Comment by Toxici11i** [comment]
this will be outclassed by dual wielding improvised crossbows, and therefor nobody would build it due to its cost

**Comment by Toxici11i** [comment]
also, 357 revolvers, mini-ebow, sketchkin, glock 17 all perform (theoretically) better than this.

**Comment by TrickyDickRichard** [comment]
> this will be outclassed by dual wielding improvised crossbows, and therefor nobody would build it due to its cost

retard

**Comment by Toxici11i** [comment]
> > this will be outclassed by dual wielding improvised crossbows, and therefor nobody would build it due to its cost
> 
> retard

try it, aim for head, duel wield at max charge. instant crit even through elite hardsuit

**Comment by diegoxt45** [comment]
is this still a thing?

**Comment by ghost** [comment]
Sir Head Maintenancer Diegoxt45, 
Please take my humblest apologies for not delivering on this Request o' the Pull. Your lowly subject has been busy with finals.  I promise to double the codemonkeying tithes for the next quarter,  your majesty.

**Comment by Toxici11i** [comment]
> Sir Head Maintenancer Diegoxt45,
> Please take my humblest apologies for not delivering on this Request o' the Pull. Your lowly subject has been busy with finals. I promise to double the codemonkeying tithes for the next quarter, your majesty.

based

**Comment by Toxici11i** [comment]
faggot

**Comment by Toxici11i** [comment]
> is this still a thing?

faggot

**Comment by diegoxt45** [comment]
> Sir Head Maintenancer Diegoxt45,
> Please take my humblest apologies for not delivering on this Request o' the Pull. Your lowly subject has been busy with finals. I promise to double the codemonkeying tithes for the next quarter, your majesty.

king

**Comment by diegoxt45** [comment]
> > is this still a thing?
> 
> faggot

i'l leave your spine in splinters

**Comment by ghost** [comment]
is this still a thing?

**Comment by yoyobatty** [comment]
Closing for now, you can reopen this anytime it's ready.

**Pull request closed without merging**